awesomeversion/awesomeversion.py

```diff
@@ -14,17 +15,18 @@
     version_strategy,
 )
 from .strategy import AwesomeVersionStrategy
+from .utils import get_regex_match_group
 
 _LOGGER: logging.Logger = logging.getLogger(__package__)
 
 
 class _AwesomeVersionBase(str):
     """Base class for AwesomeVersion to allow the usage of the default JSON encoder."""
 
-    def __init__(self, string):
+    def __init__(self, string: str):
         str.__init__(string)
 
-    def __new__(cls, version, _=None):
+    def __new__(cls, version: str, _: Optional[Any] = None) -> "_AwesomeVersionBase":
```

Comment:
You can add `from __future__ import annotation` to use unquoted type.
```suggestion
    def __new__(cls, version: str, _: Optional[Any] = None) -> _AwesomeVersionBase:
```

Reply:
`from __future__ import annotation` can only be used from python 3.8, this library has support for 3.6 and 3.7 as well.